python/ray/serve/tests/test_task_processor.py

```diff
@@ -0,0 +1,223 @@
+    This fixture uses 'yield' to ensure the directory persists for the duration of the test.
+
+    Yields:
+        dict: Contains 'queue_path', 'results_path', and 'control_path' for easy access
+    """
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        tmpdir_path = Path(tmpdir)
+
+        data_folder_queue = tmpdir_path / "queue"
+        data_folder_queue.mkdir()
+
+        results_path = tmpdir_path / "results"
+        results_path.mkdir()
+
+        control_path = tmpdir_path / "control"
+        control_path.mkdir()
+
+        yield {
+            "queue_path": data_folder_queue,
+            "results_path": results_path,
+            "control_path": control_path,
+        }
+
+
+class TestTaskConsumerWithRayServe:
+    """Test task consumer integration with Ray Serve."""
+
+    DEFAULT_QUEUE_NAME = "my_default_app_queue"
+    DEFAULT_BROKER_URL = "filesystem://"
+
+    @classmethod
+    def _create_transport_options(cls, queue_path, control_path):
+        """Create standard transport options for filesystem broker."""
+        return {
+            # Incoming message queue - where new task messages are written when sent to broker
+            "data_folder_in": queue_path,
+            # Outgoing message storage - where task results and responses are written after completion
+            "data_folder_out": queue_path,
+            # Processed message archive - where messages are moved after successful processing
+            "data_folder_processed": queue_path,
+            # Control message storage - where Celery management and control commands are stored
+            "control_folder": control_path,
+        }
+
+    @classmethod
+    def _create_processor_config(cls, temp_queue_directory, **kwargs):
+        """Create a TaskProcessorConfig with common defaults."""
+        queue_path = temp_queue_directory["queue_path"]
+        control_path = temp_queue_directory["control_path"]
+        results_path = temp_queue_directory["results_path"]
+
+        transport_options = cls._create_transport_options(queue_path, control_path)
+
+        config_params = {
+            "queue_name": cls.DEFAULT_QUEUE_NAME,
+            "adapter_config": CeleryConfig(
+                broker_url=cls.DEFAULT_BROKER_URL,
+                backend_url=f"file://{results_path}",
+                broker_transport_options=transport_options,
+            ),
+        }
+        config_params.update(kwargs)
+
+        return TaskProcessorConfig(**config_params)
+
+    @staticmethod
+    def _create_send_request_to_queue_remote(processor_config):
+        """Create a Ray remote function for sending requests to the queue."""
+
+        @ray.remote
+        def send_request_to_queue(data):
+            celery_adapter = get_task_adapter(config=processor_config)
+            result = celery_adapter.enqueue_task_sync("process_request", args=[data])
+            assert result.id is not None
+            return result.id
+
+        return send_request_to_queue
```

Comment:
its a little odd to wrap the ray task into a function. Instead I suggest keeping utility at top level outside the class that way it can be used by other test files and reads a little cleaner

```
        @ray.remote
        def send_request_to_queue(processor_config, task_name, *args, **kwargs):
            celery_adapter = get_task_adapter(config=processor_config)
            result = celery_adapter.enqueue_task_sync(task_name, args=args, kwargs=kwargs)
            assert result.id is not None
            return result.id

def test_myfunc():
   ...
   send_request_to_queue.remote(....)
```